### ISSUE TITLE
->toOthers() modifier should be per session, not per user

### ISSUE THREAD

**Issue by SirLamer** [laravel/echo#105]
In the case where one user has multiple browser tabs open, among which some should receive event updates from activity in another tab, the "toOthers()" modifier will result in those other local tabs not being updated. I suspect the same is true if a user is logged in on multiple separate devices. I suggest that "toOthers()" is really meant to exclude the specific browser session from which the event was received, not really meant to exclude all cases of the given user, and probably an alternate modifier with the behavior above should be provided (different method name for backwards compatibility reasons). In its current state my feeling is that this function is somewhat broken in its current form.

Testing on Laravel 5.3

**Comment by hinaloe** [comment]
Really? `toOthers` is using `socketId`, and it generate per session by server (ex [socket.io](https://socket.io/docs/client-api/#socket), or pusher).

So I think it may send other session of same user...

**Comment by SirLamer** [comment]
When I tested it in March as you describe, it worked as describe in the OP.

**Comment by d-lamers** [comment]
Having the same issue here, here are my findings:
`toOthers()` indeed uses the header `X-Socket-ID` and I make sure it is set. I even see that the socket is send with the event. But the client with that socketId still gets the event. I use socketio as driver.

**Comment by SirLamer** [comment]
I have investigated this further, and found that because I was using an off-spec client-side API engine, I was not including "X-Socket-ID" in the request. Because it was missing, Laravel handled the absence gracefully and just broadcast to everyone by omitting the Socket ID from the event, in my case sent to Pusher.

https://laravel.com/docs/5.5/broadcasting#broadcasting-events

In the debug console on Pusher, if you are watching for events and the "socket" column is blank on new events, it will be sent to everyone. If the "socket" includes your client's socket ID, it that client (per session, not per user) will be omitted.

ALSO IMPORTANT: If you call Echo.socketId() during script initialization and then store it, the socket ID will not be available yet so this returns "undefined". In my case I prepared a simple middleware which retrieves the socket ID from Echo during each API call.

Example (using JSON Devour: https://github.com/twg/devour)
```js
import JsonApi from 'devour-client/src';
import Echo from '../echo'

const jsonApi = new JsonApi({apiUrl: '/api/v1'});
jsonApi.headers['Accept'] = 'application/vnd.api+json';
jsonApi.headers['X-CSRF-TOKEN'] = window.Laravel.csrfToken;
jsonApi.headers['X-Requested-With'] = 'XMLHttpRequest';

const getSocketId = {
    name: 'get-socket-id',
    req: (payload) => {
        payload.req.headers['X-Socket-ID'] = Echo.socketId();

        return payload;
    }
};

// More stuff

jsonApi.insertMiddlewareBefore('axios-request', getSocketId);

// More stuff

export default jsonApi;
```
